src/Parser/DeathKnight/Blood/Modules/Features/BoneShieldUptime.js

```diff
@@ -1,20 +1,40 @@
 import React from 'react';
-
 import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
 import SPELLS from 'common/SPELLS';
 import SpellIcon from 'common/SpellIcon';
 import { formatPercentage } from 'common/format';
 import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
 
 class BoneShieldUptime extends Analyzer {
+
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  get Uptime() {
+    return this.combatants.getBuffUptime(SPELLS.BONE_SHIELD.id) / this.owner.fightDuration;
```

Comment:
This can only be cast on the player right? I think it would be better to use `this.combatants.selected`.